lib/endpoints/vms.js

```diff
@@ -213,6 +218,18 @@ function listVms(req, res, next) {
 
             res.header('x-joyent-resource-count', count);
 
+            var limit = DEFAULT_LIST_VM_LIMIT;
+            if (req.params.limit)
+                limit = Number(req.params.limit);
+
+            params.limit = Math.min(limit, MAX_LIST_VMS_LIMIT);
```

Comment:
I think we should probably be checking that this number is within the
valid range in the `validateListVmsParams()` function.  Could a min/max
be added to `createValidateNumberFn()` so that we can provide clear
feedback to the consumer about any invalid input?